text/trace/0168-sampling-propagation.md

```diff
@@ -0,0 +1,454 @@
+corresponds with being sampled at probabilities 1-in-1 through 1-in-8
+but not at probabilities 1-in-16 and smaller.
+
+### Proposed `tracestate` syntax
+
+The consistent sampling r-value (`r`) and and head sampling
+probability p-value (`p`) will be propagated using two bytes of base16
+content for each of the two fields, as follows:
+
+```
+tracestate: otel=p:PP;r:RR
+```
+
+where `PP` are two bytes of base16 p-value and `RR` are two bytes of
+base16 r-value.  These values are omitted when they are unknown.
+
+This proposal should be taken as a recommendation and will be modified
+to [match whatever format OpenTelemtry specifies for its
+`tracestate`](https://github.com/open-telemetry/opentelemetry-specification/pull/1852).
+The choice of base16 encoding is therefore just a recommendation,
+chosen because `traceparent` uses base16 encoding.
+
+### Examples
+
+The following `tracestate` value:
+
+```
+tracestate: otel=r:0a;p:03
+```
+
+translates to
+
+```
+base16(p-value) = 03 // 1-in-4 head probability
+base16(r-value) = 0a // qualifies for 1-in-1024 or greater probability consistent sampling
+```
+
+Any `TraceIDRatioBased` Sampler configured with probability 2**-10 or
+greater will enable sampling this trace, whereas any
+`TraceIDRatioBased` Sampler configured with probability 2**-11 or less
+will stop sampling this trace.
+
+## Internal details
+
+The reasoning behind restricting the set of sampling rates is that it:
+
+- Lowers the cost of propagating head sampling probability
+- Limits the number of random bits required
+- Avoids floating-point to integer rounding errors
+- Makes math involving partial traces tractable.
+
+[An algorithm for making statistical inference from partially-sampled
+traces has been published](https://arxiv.org/pdf/2107.07703.pdf) that
+explains how to work with a limited number of power-of-2 sampling rates.
+
+### Behavior of the `TraceIDRatioBased` Sampler
+
+The Sampler MUST be configured with a power-of-two probability
+expressed as `2**-s` with s being an integer in the range [0, 61]
+except for the special case of zero probability.
+
+If the context is a new root, the initial `tracestate` must be created
+with randomness value `r`, as described above, in the range [0, 61].
+If the context is not a new root, output a new `tracestate` with the
+same `r` value as the parent context.
+
+When sampled, in both cases, the context's p-value `p` is set to the
+value of `s+1` in the range [1, 62].  If the sampling probability is
+zero (the special case where `s` is undefined), use `p=63` the
+specified value for zero probability.
+
+In both cases, set the sampled bit if the outgoing `p` minus one is
+less than or equal to the outgoing `r` (i.e., `p-1 <= r`).
+
+If the context is not a new root and the incoming context's r-value
+is not set, the implementation SHOULD notify the user of an error
+condition and follow the incoming context's `sampled` flag.
```

Comment:
This statement makes this feature "required" for all deployments, which I think it should not be.

Reply:
This level of detail about specific error handling, I think, can be debated in https://github.com/open-telemetry/opentelemetry-specification/pull/1899.

Reply:
Please make sure the comment is capture there.